libs/execution_context/ebpf_maps.c

```diff
@@ -309,17 +378,96 @@ ebpf_map_function_table_t ebpf_map_function_tables[] = {
     {// BPF_MAP_TYPE_UNSPECIFIED
      NULL},
     {// BPF_MAP_TYPE_HASH
-     ebpf_create_hash_map,
-     ebpf_delete_hash_map,
-     ebpf_find_hash_map_entry,
-     ebpf_update_hash_map_entry,
-     ebpf_delete_hash_map_entry,
-     ebpf_next_hash_map_key},
+     _create_hash_map,
+     _delete_hash_map,
+     _find_hash_map_entry,
+     _update_hash_map_entry,
+     NULL,
+     _delete_hash_map_entry,
+     _next_hash_map_key},
     {// BPF_MAP_TYPE_ARRAY
-     ebpf_create_array_map,
-     ebpf_delete_array_map,
-     ebpf_find_array_map_entry,
-     ebpf_update_array_map_entry,
-     ebpf_delete_array_map_entry,
-     ebpf_next_array_map_key},
+     _create_array_map,
+     _delete_array_map,
+     _find_array_map_entry,
+     _update_array_map_entry,
+     NULL,
+     _delete_array_map_entry,
+     _next_array_map_key},
+    {// BPF_MAP_TYPE_PROG_ARRAY
+     _create_prog_array_map,
+     _delete_array_map_with_references,
```

Comment:
should all these have prog_array in the name so that it is more prominent that these are implementations for the program_array type?

Reply:
they intentionally do not since the same functions can be used with BPF_MAP_TYPE_ARRAY_OF_MAPS once we implement that.